File: CreationalPatterns/AbstractFactory/Java/Vehicle/src/concretefactories/ItalianVehicleFactory.java
package concretefactories;

import interfaces.IVehicle;
import interfaces.IVehicleFactory;
import products.ItalianSUV;
import products.ItalianSedan;
import products.ItalianSportsCar;

public class ItalianVehicleFactory implements IVehicleFactory {

    @Override
    public IVehicle createSedan() {
        return new ItalianSedan();
    }

    @Override
    public IVehicle createSUV() {
        return new ItalianSUV();
    }

    @Override
    public IVehicle createSportsCar() {
        return new ItalianSportsCar();
    }
}

File: CreationalPatterns/AbstractFactory/C#/Vehicle Factory/ConcreteFactories/ItalianVehicleFactory.cs
using Vehicles.Interfaces;
using Vehicles.Products;

namespace Vehicles.ConcreteFactories
{
    public class ItalianVehicleFactory : IVehicleFactory
    {
        public IVehicle CreateSedan()
        {
            return new ItalianSedan();
        }

        public IVehicle CreateSUV()
        {
            return new ItalianSUV();
        }

        public IVehicle CreateSportsCar()
        {
            return new ItalianSportsCar();
        }
    }
}